docs/modules/ROOT/pages/explanations/decisions/mariadb-proxy.adoc

```diff
@@ -0,0 +1,162 @@
+
+|Pricing
+|Only available with MariaDB Enterprice
+|Free (Paid support available)
+|Free (Paid enterprice version available)
+
+|===
+
+
+=== Analysis - MariaDB Maxscale
+
+https://mariadb.com/kb/en/maxscale/[MariaDB Maxscale]
+
+This is the official proxy solution by MariaDB itself.
+It is by far the feature richest solution. However, this comes at a hefty price.
+Maxscale comes bundled with MariaDB Enterprice and is not available as a standalone solution.
+MariaDB doesn't publish any pricing lists, however research showed that MariaDB Enterprice for three nodes is in the lower 5 figures per year.
+
+Advantages::
+
+* Written and maintained by the MariaDB company
+* Layer 4 (SQL aware proxy)
+* Huge amount of features (including Read-write splitting)
+* Easy to configure
+* Support from MariaDB through Enterprise subscription
+
+Disadvantages::
+
+* BSL
+* Only available through Enterprise subscription
+* Licensing tracking required
+* Slight performance impact due to being a Layer 4 proxy
+* No official helm chart
+
+=== Analysis - ProxySQL
+
+|https://proxysql.com/[ProxySQL]
+
+ProxySQL is a lightweight SQL proxy published under the GPL 3.0. The software itself is
+free to use. The company offers support plans if desired.
+ProxySQL is not as feature rich Maxscale, but it still has a quite big feature list. Most notably the read-write splitting.
+
+Advantages::
+
+* OpenSource GPL 3
+* Layer 4 (SQL aware proxy)
+* Read-write splitting
+* Galera Support
+* Slight performance impact due to being a Layer 4 proxy
+* Helm chart availble
+
+Disadvantages::
+
+* A bit more complex to configure than maxscale
+* We don't have experience with this tool yet
+
+=== Analysis - HAProxy
+
+|https://www.haproxy.org/[HAProxy]
+
+HAProxy is an open source multi purpose proxy and loadbalancer. It is known for it's high performance and ability to handle huge amounts of requests.
+However, HAProxy is only able to proxy and loadbalance traffic on either HTTP/S or TCP level.
+It doesn't understand SQL and hence can't load balance traffic depending on the query.
+
+Advantages::
+
+* OpenSource GPL/LGPL
+* High performance
+* We know HAProxy well
+* Helm chart
+* Easy to configure
+
+Disadvantages::
+
+* Layer 4 (TCP Proxy)
+* No Loadbalancing based on query
+
+
+== Decision
+
+We use ProxySQL
+
+== Rationale
+
+Maxscale is only available as a bundle with MariaDB Enterprise, which is too expensive for our use case and would make the whole product very hard to sell.
+
+Athough we have quite some experience with HAProxy and is performing very well, it unfortunately lacks any support for SQL query based routing / load balancing.
+This is a huge draw back and limits the usages of a galera cluster considerably as we won't the Galera nodes only during failover scenarios, but not for load balancing in general.
+
+ProxySQL is also a rather lightweight solution build for high performance. However, it can't keep up with the performance of the Layer 3 HAProxy solution.
```

Comment:
>  However, it can't keep up with the performance of the Layer 3 HAProxy solution.

Are there some benchmarks? Or did you do your benchmarks yourself?

Reply:
I've added a footnote to some benchmarks done by percona